javascript/ql/src/semmle/javascript/dataflow/Configuration.qll

```diff
@@ -633,7 +633,12 @@ private predicate exploratoryFlowStep(
  */
 private predicate isSource(DataFlow::Node nd, DataFlow::Configuration cfg, FlowLabel lbl) {
   (cfg.isSource(nd) or nd.(AdditionalSource).isSourceFor(cfg)) and
-  lbl = FlowLabel::data()
+  (
+    if cfg instanceof TaintTracking::Configuration then
```

Comment:
It is a shame that we have to introduce a dependency form `Configuration.qll` to `TaintTracking.qll` (I think this is the first, at least). We could avoid that by introducing `cfg.getDefaultSourceLabel()`, which in `TaintTracking::configuration` is implemented as `result = FlowLabel::taint()`. 
I will leave it to you to decide what to do here.

Ditto for `isSink` below.

Reply:
That's very true, but on the other hand, there's also a downside to extending our public API with stuff that users don't actually need.